pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -147,20 +149,35 @@ func parseConfig(getconfigCtx *getconfigContext, config *zconfig.EdgeDevConfig,
 		parseSystemAdapterConfig(getconfigCtx, config, source, forceSystemAdaptersParse)
 
 		if source != fromBootstrap {
-			parseBaseOS(getconfigCtx, config)
+			activateNewBaseOSFlag := parseBaseOS(getconfigCtx, config)
 			parseNetworkInstanceConfig(getconfigCtx, config)
 			parseContentInfoConfig(getconfigCtx, config)
 			parseVolumeConfig(getconfigCtx, config)
 
-			// parseProfile must be called before processing of app instances from config
-			parseProfile(getconfigCtx, config)
-			parseAppInstanceConfig(getconfigCtx, config)
+			// If we are activating a new baseOS, we need to skip processing
+			if source == fromController && activateNewBaseOSFlag {
```

Comment:
In addition you need to check whether a previous parseBaseOs returned true.

Changing the return value from bool to the enum makes that easy if you change the getconfigCtx.configProcessingSkipFlagReboot to of the enum type (and different name); basically save the previous configProcessingRetval in getconfigCtx, Any time that changes the configTimerTask can triggerPubishDevinfo etc.

Reply:
I addressed the commend in a newer version. I think it's fine now.